### PR TITLE
PP-6708 Upgrade to JUnit5

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT
- Upgrades unit tests to Junit5.

## HOW
- Units tests should pass on CI

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -172,6 +172,12 @@
             <version>1.1.1</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-params</artifactId>
+            <version>${junit5.version}</version>
```
Comment (alexbishop1) [inline pom.xml:178]: Where is this being defined?

File: src/test/java/uk/gov/pay/ledger/event/service/EventServiceTest.java [hunk 2]
```diff
@@ -55,7 +56,7 @@ public void setUp() {
                 .withResourceExternalId(resourceExternalId)
                 .withEventDate(ZonedDateTime.now().minusHours(2L))
                 .toEntity();
-        when(mockEventDao.getEventsByResourceExternalId(resourceExternalId)).thenReturn(List.of(event1, event2));
+        lenient().when(mockEventDao.getEventsByResourceExternalId(resourceExternalId)).thenReturn(List.of(event1, event2));
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/ledger/event/service/EventServiceTest.java:59]: Why is this needed now?

File: pom.xml (hunk 1 above)
Comment (kbottla) [reply pom.xml:178]: It is using the one defined in parent POM (dropwizard). Updated in this PR https://github.com/alphagov/pay-ledger/pull/822/files#diff-600376dffeb79835ede4a0b285078036R18

File: src/test/java/uk/gov/pay/ledger/event/service/EventServiceTest.java (hunk 2 above)
Comment (kbottla) [reply src/test/java/uk/gov/pay/ledger/event/service/EventServiceTest.java:59]: Mockito for Junit5 complains about unused stubbing and tests fails, as this stub is only used for few tests. It might be that previous version doesn't require strict stubs or has different rules for it. Going to look into more detail
